src/menu/MenuSongSelection.as

```diff
@@ -1476,240 +1801,268 @@ package menu
                     if (!songItems[newIndex].isLocked)
                     {
                         setActiveIndex(newIndex, lastIndex, true);
-                        buildInfoTab();
+                        buildInfoBox();
                         break;
                     }
                 }
             }
         }
 
-        private function songLoadClick(e:Event):void
-        {
-            multiplayerLoad(e.target.level);
-        }
-
-        private function multiplayerLoad(level:int):void
+        /**
+         * Mouse Wheel Handler for the Playlist Scroll Pane.
+         * Moves the scroll pane based on the scroll delta direction.
+         * @param e
+         */
+        private function mouseWheelHandler(e:MouseEvent):void
         {
-            if (level < 0)
+            //- Sanity
+            if (genreLength == 0 || !scrollbar.draggerVisibility)
                 return;
 
-            isQueuePlaylist = false;
-            _mp.gameplayPicking(_playlist.getSong(level));
-            _mp.gameplayLoading();
-            switchTo(MainMenu.MENU_MULTIPLAYER);
+            //- Scroll
+            options.scroll_position = scrollbar.scroll + (pane.scrollFactorVertical / 2) * (e.delta > 0 ? -1 : 1);
+            pane.scrollTo(options.scroll_position);
+            scrollbar.scrollTo(options.scroll_position);
         }
 
-        private function playSong(level:int):void
+        /**
+         * Scroll Bar Moved Update Handler for the Playlist Scroll Bar.
+         * Updates the scroll pane postion based on the scroll bar position.
+         * @param e
+         */
+        private function scrollBarMoved(e:Event):void
         {
-            if (level < 0)
-                return;
+            options.scroll_position = e.target.scroll;
+            pane.scrollTo(e.target.scroll);
+        }
 
-            _gvars.songQueue = [];
-            var songData:Object = _playlist.getSong(level);
-            if (songData.error == null)
-            {
-                _gvars.songQueue.push(songData);
-                playQueue();
-            }
+        //******************************************************************************************//
+        // Menu Music
+        //******************************************************************************************//
+
+        /**
+         * Callback for Menu Music Song when loaded and available.
+         */
+        private function e_menuMusicConvertSongLoad(e:Event):void
+        {
+            writeMenuMusicBytes(e.target as Song);
+            playMenuMusicSong(e.target as Song);
         }
```

Comment:
```suggestion
        private function e_menuMusicConvertSongLoad(e:Event):void
        {
            var song:Song = (e.target as Song);
            song.removeEventListener(Event.COMPLETE, e_menuMusicConvertSongLoad);
            writeMenuMusicBytes(song);
            playMenuMusicSong(song);
        }
```
Not sure if this is correct, but as the `e_menuMusicConvertSongLoad` event listener doesn't use a weak reference, the listener wouldn't get garbage collected automatically, so there should be some method to remove it.